client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js

```diff
@@ -24,14 +24,14 @@ export default function ThemeTierPartnerBadge() {
 	const subscriptionPrices = useSelector( ( state ) =>
 		getMarketplaceThemeSubscriptionPrices( state, themeId )
 	);
-	const { isThemeAllowedOnSite } = useThemeTier( siteId, themeId );
+	const isThemeAllowed = useSelector( ( state ) => isThemeAllowedOnSite( state, themeId ) );
 
-	const labelText = isThemeAllowedOnSite
+	const labelText = isThemeAllowed
```

Comment:
Nitpick: if we keep the `isThemeAllowed` const, we don't need to rename it in multiple places.